test/jsx-no-logical-expression.js

```diff
@@ -0,0 +1,87 @@
+import test from 'ava';
+import avaRuleTester from 'eslint-ava-rule-tester';
+import ruleFixer from 'eslint/lib/linter/rule-fixer';
+import rule, {createFixer} from '../src/rules/jsx-no-logical-expression';
+
+const ruleTester = avaRuleTester(test, {
+  parserOptions: {
+    ecmaVersion: 2018,
+    ecmaFeatures: {
+      jsx: true
+    }
+  }
+});
+
+const ruleId = 'jsx-no-logical-expression';
+const message = 'JSX should not use logical expression';
+
+const error = {
+  ruleId,
+  message,
+  type: 'LogicalExpression'
+};
+
+ruleTester.run(ruleId, rule, {
+  valid: [
+    '{true ? <div /> : null}',
+    '{false || false ? <div /> : null}',
+    '{true && true ? <div /> : null}'
+  ],
+  invalid: [
+    {
+      code: '{true && <div />}',
+      errors: [error]
+    },
+    {
+      code: '{true || <div />}',
+      errors: [error]
+    },
+    {
+      code: '{false && <div />}',
+      errors: [error]
+    },
+    {
+      code: '{undefined && <div />}',
+      errors: [error]
+    },
+    {
+      code: '{0 && <div />}',
+      errors: [error]
+    }
+  ]
+});
+
+const createNode = text => {
+  const operatorPosition = text.search(/(&&|\|\|)/g);
+  const operator = text.slice(operatorPosition, operatorPosition + 2);
+  return {
+    start: 0,
+    end: text.length,
+    range: [0, text.length],
+    operator,
+    left: {
+      end: operatorPosition - 1
+    },
+    right: {
+      start: operatorPosition + 3
+    }
+  };
+};
+
+const testFixes = (t, text, fixes) => {
```

Comment:
Instead of having tests specifically for the fixes, ESLint allows to specify an `output` property in the test which will assert the fix's result in a more readable way (it's easier to see if the result will still be valid JS).

Example:
```js
    {
      code: '{true || <div />}',
      output: '{true ? <div /> : null}',
      errors: [error]
    }
```

I suggest adding the `output` property in every `invalid` test case. It's unfortunately not mandatory, but it's highly useful.


Oh, by the way, hi!

![](https://media.giphy.com/media/l0FF56cexcW2JAXCJj/giphy.gif)

Reply:
![https://media.giphy.com/media/l3V0oW9jbIk7CWpsA/source.gif](https://media.giphy.com/media/l3V0oW9jbIk7CWpsA/source.gif)

Hi @jfmengels,

Thx for your review, that was exactly what i was searching for 👍